eng/testing/AppleRunnerTemplate.sh

```diff
@@ -39,11 +32,26 @@ while true; do
     fi
 done
 
-XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
-
-dotnet xharness ios test --app="$APP_BUNDLE" \
-    --targets=$TARGET \
-    --output-directory=$XHARNESS_OUT
+export XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
+
+if [ -x "$(command -v xharness)" ]
+then
+    echo 'Xharness command is in $PATH'
+    export XHARNESS_DISABLE_COLORED_OUTPUT=true
+    export XHARNESS_LOG_WITH_TIMESTAMPS=true
+    xharness ios test \
+        --targets="$TARGET" \
+        --app="$APP_BUNDLE" \
+        --output-directory=$XHARNESS_OUT
+    echo "Output files:"
+    ls -lh $XHARNESS_OUT
+else
+    echo 'Xharness command is NOT in $PATH'
+    dotnet xharness ios test \
+        --targets="$TARGET" \
+        --app="$APP_BUNDLE" \
+        --output-directory=$XHARNESS_OUT
+fi
 
 _exitCode=$?
```

Comment:
I think you are exiting with the exit code of `ls -lh $XHARNESS_OUT` and then the job succeeds even when XHarness fails